Remove submitted task from task list

nbarnabee opened the pull request:
This PR addresses [T331269](https://phabricator.wikimedia.org/T331269)

Upon a "successful" task submission (e.g., one that does not return an error on the front end), the submitted task will be removed from the task array and a new one will be presented.

I did make a pretty significant change to the Homepage structure:  as I needed to emit directly from UserContributionForm to Homeview, I stripped out the Home.vue component entirely and moved its contents to HomeView.vue.  

An alternative would be to restore Home.vue, but as a sibling component to the UserContributionForm, rather than having a chained set of child components.  (That's why I've left Home.vue in the file system rather than removing it.)

nbarnabee commented:
Hmm, except that now the tests are failing.  Argh.

Damilare1 commented on src/components/__tests__/Home.spec.js:
```diff
@@ -97,18 +55,18 @@ describe("Home", () => {
-      "https://toolsadmin.wikimedia.org/tools/id/authors"
-    );
-    expect(rows[3].findAll("td")[1].text()).toBe("openhub_id");
+    //   // Expect the table to be updated with the next task details
```
We shouldn't merge commented out code. Do we want to dispose off the entire test or can we fix it?

nbarnabee replied:
The current test will at least test that, when given certain props, the table will render correctly.  So it's not entirely without value, but a test for HomeView would be more useful.

nbarnabee replied:
Addressed with [b3895cc](https://github.com/wikimedia/toolhunt-ui/pull/41/commits/b3895ccc14df0fc0d4bc2bec6257fa73d9fa4ecb)

Damilare1 commented on src/views/HomeView.vue:
```diff
@@ -1,21 +1,66 @@
-    isError.value = true
-    console.log(error)
+    tasks.value = await getTasks();
+    currentTask.value = tasks.value[currentTaskIndex.value];
```
Looks like this is repeated on line 26, makes me wonder why we had to change the watchEffect to onMounted?

nbarnabee replied:
You're right, this line was unnecessary.

nbarnabee replied:
Removed in [b3895cc](https://github.com/wikimedia/toolhunt-ui/pull/41/commits/b3895ccc14df0fc0d4bc2bec6257fa73d9fa4ecb)

Damilare1 commented on src/components/Home.vue:
```diff
@@ -1,31 +1,21 @@
 <script setup>
-import { ref, defineProps, watchEffect } from "vue";
-import UserContributionForm from "./UserContributionForm.vue";
+// import { stringify } from "querystring";
```
Let's not merge commented out code, we can delete it now and add it back when we need it.

nbarnabee replied:
Addressed in [b3895cc](https://github.com/wikimedia/toolhunt-ui/pull/41/commits/b3895ccc14df0fc0d4bc2bec6257fa73d9fa4ecb)

Damilare1 commented on src/components/Home.vue:
```diff
@@ -1,31 +1,21 @@
-    currentTaskIndex.value = 0;
-  }
-}
+// const currentTaskIndex = ref(0);
```
No commented out code

nbarnabee replied:
Addressed in [b3895cc](https://github.com/wikimedia/toolhunt-ui/pull/41/commits/b3895ccc14df0fc0d4bc2bec6257fa73d9fa4ecb)

Damilare1 reviewed (commented):
Thanks for all the work on this @nbarnabee. 

Looks like there's a lot happening on this PR. We need to discuss this process and would be good to have some good unit test support to it too. We can focus on this after the first deployment.

nbarnabee commented:
Something seems to have gone funny when I merged the changes from main into this branch; I'm getting warnings that I didn't before and haven't yet gotten to the bottom of it, though everything still seems to be rendering correctly.

nbarnabee commented:
[b3895cc](https://github.com/wikimedia/toolhunt-ui/pull/41/commits/b3895ccc14df0fc0d4bc2bec6257fa73d9fa4ecb) addresses comments made by Damilare.  I still haven't resolved the warnings I alluded to in my earlier messages, but it looks like this change isn't going in prior to deployment anyway.  I will have another look at it later.

nbarnabee commented:
Given the numerous changes to the frontend, this will probably require some re-working.  (And I STILL haven't figured out what was causing those errors.)  If anyone else would like to take a stab at this, feel free.

nbarnabee commented:
Reopening this one.  I merged all of the changes that had happened on the main branch since the last time I updated it, and am no longer getting the various warnings.  Functionally, this does what it's supposed to.  

I still do not have a full test of everything that passes from HomeView to the child components and don't have an ETA on that.

nbarnabee commented:
Hey @Damilare1, I've fixed the merge conflicts.
